profiles/events.py

```diff
@@ -57,7 +57,7 @@ def event_handler(sender: Any, changes: List[Tuple[db.Model, str]]) -> None:
             if isinstance(instance, Profile):
                 ids.append(instance.id)
             if isinstance(instance, (Affiliation, EmailAddress)):
-                ids.append(instance.profile.id)
+                ids.append(instance.profile_id)
```

Comment:
(From previous discussion)

You lose the Profile model relationship reference from the EmailAddress or Affiliation object instance(s) passed to the event handlers on deletion, whereas they are available on update and insert. That's why 'instance.profile` was `None` and `instance.profile_id` was available.

Reply:
is `profile_id` also magically set by the ORM? I can't see an assignment for it

Reply:
Yep.